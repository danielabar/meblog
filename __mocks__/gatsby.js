const React = require("react")
const gatsby = jest.requireActual("gatsby")

module.exports = {
  ...gatsby,
  graphql: jest.fn(),
  Link: jest.fn().mockImplementation(
    // these props are invalid for an `a` tag
    ({
      activeClassName,
      activeStyle,
      getProps,
      innerRef,
      partiallyActive,
      ref,
      replace,
      to,
      ...rest
    }) =>
      React.createElement("a", {
        ...rest,
        href: to,
      })
  ),
  Slice: jest.fn().mockImplementation(
    ({ alias, ...rest }) =>
      React.createElement("div", {
        ...rest,
        "data-test-slice-alias": alias
      })
  ),
  // TODO: Remove, no longer using StaticQuery component
  StaticQuery: jest.fn().mockImplementation(() => {
    return {
      site: {
        siteMetadata: {
          defaultTitle: "Jane Doe Blog",
          titleTemplate: "%s · Jane Doe",
          defaultDescription: "Blog description.",
          siteUrl: "https://someblog.com",
          defaultImage: "/images/profile.png",
          twitterUsername: "@somebody",
          googleSiteVerification: "abc123",
        },
      },
    }
  }),
  // This works because only SEO component uses useStaticQuery hook
  useStaticQuery: jest.fn().mockImplementation(() => {
    return {
      site: {
        siteMetadata: {
          defaultTitle: "Jane Doe Blog",
          titleTemplate: "%s · Jane Doe",
          defaultDescription: "Blog description.",
          siteUrl: "https://someblog.com",
          defaultImage: "/images/profile.png",
          twitterUsername: "@somebody",
          googleSiteVerification: "abc123",
        },
      },
    }
  }),
  navigate: jest.fn(),
}
