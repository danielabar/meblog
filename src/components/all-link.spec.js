import React from "react"
import renderer from "react-test-renderer"

import AllLink from "./all-link"
import { Link } from "gatsby"

describe("AllLink", () => {
  it("renders correctly", () => {
    const testRenderer = renderer.create(<AllLink marginTop="30px"/>);
    expect(testRenderer.toJSON()).toMatchSnapshot()

    const testInstance = testRenderer.root;
    expect(testInstance.findByType(Link)).toBeDefined();
  })
})