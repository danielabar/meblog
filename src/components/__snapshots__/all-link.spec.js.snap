// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AllLink renders correctly 1`] = `
<div
  style={
    Object {
      "marginTop": "30px",
    }
  }
>
  <a
    className="allLink"
    href="/blog"
  >
    All Articles
  </a>
</div>
`;
